Test compilation failure with entity generated with service option

##### **Overview of the issue**

Test compilation failure:

> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project jhipster: Compilation failure: Compilation failure:
[ERROR] /C:/dev/workspace/jhipster/tests/mn-mvn-jwt-angular/src/test/java/com/mycompany/myapp/web/rest/LabelResourceIT.java:[165,9] cannot find symbol
[ERROR]   symbol:   variable labelService
[ERROR]   location: class com.mycompany.myapp.web.rest.LabelResourceIT
[ERROR] /C:/dev/workspace/jhipster/tests/mn-mvn-jwt-angular/src/test/java/com/mycompany/myapp/web/rest/LabelResourceIT.java:[210,9] cannot find symbol
[ERROR]   symbol:   variable labelService
[ERROR]   location: class com.mycompany.myapp.web.rest.LabelResourceIT


##### **Reproduce the error**

Generate an entity with serviceClass or serviceImpl value for service option.

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

Always save the test entity with the entity repository, no need to conditionally use the entity service.

##### **Project configuration**

<!--
To provide all information we need, you should run `mhipster info` or `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
---
If you have a JDL please wrap it in below structure
  <details>
  <summary>JDL definitions</summary>
  <pre>
     JDL content here
  </pre>
  </details>
-->

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

```json
{
  "name": "Label",
  "fields": [
    {
      "fieldName": "label",
      "fieldType": "String",
      "fieldValidateRules": ["required", "minlength"],
      "fieldValidateRulesMinlength": "3"
    }
  ],
  "relationships": [
    {
      "relationshipType": "many-to-many",
      "otherEntityName": "operation",
      "otherEntityRelationshipName": "label",
      "relationshipName": "operation",
      "otherEntityField": "id",
      "ownerSide": false
    }
  ],
  "changelogDate": "20201020221801",
  "entityTableName": "label",
  "dto": "no",
  "pagination": "no",
  "service": "serviceClass",
  "jpaMetamodelFiltering": false,
  "fluentMethods": true,
  "readOnly": false,
  "embedded": false,
  "clientRootFolder": "",
  "applications": ["jhipster"]
}

```


##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->

-   [X] Checking this box is mandatory (this is just to show you read everything)

--- a/generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs
+++ b/generators/entity-server/templates/src/test/java/package/web/rest/EntityResourceIT.java.ejs
@@ -521,15 +521,11 @@ public class <%= entityClass %>ResourceIT <% if (databaseType === 'cassandra') {
         <%_ if (databaseType === 'cassandra') { _%>
         <%= asEntity(entityInstance) %>.setId(UUID.randomUUID());
         <%_ } _%>
-<%_ if (service !== 'no' && dto !== 'mapstruct') { _%>
-        <%= entityInstance %>Service.save(<%= asEntity(entityInstance) %>);
     <%_ if (searchEngine === 'elasticsearch') { _%>
         // As the test used the service layer, reset the Elasticsearch mock repository
         reset(mock<%= entityClass %>SearchRepository);
     <%_ } _%>
-<%_ } else { _%>
         <%= entityInstance %>Repository.save<% if (databaseType === 'sql') { %>AndFlush<% } %>(<%= asEntity(entityInstance) %>);
-<%_ } _%>
 
         int databaseSizeBeforeUpdate = <%= entityInstance %>Repository.findAll().size();
 
@@ -595,11 +591,7 @@ public class <%= entityClass %>ResourceIT <% if (databaseType === 'cassandra') {
         <%_ if (databaseType === 'cassandra') { _%>
         <%= asEntity(entityInstance) %>.setId(UUID.randomUUID());
         <%_ } _%>
-<%_ if (service !== 'no' && dto !== 'mapstruct') { _%>
-        <%= entityInstance %>Service.save(<%= asEntity(entityInstance) %>);
-<%_ } else { _%>
         <%= entityInstance %>Repository.save<% if (databaseType === 'sql') { %>AndFlush<% } %>(<%= asEntity(entityInstance) %>);
-<%_ } _%>
 
         int databaseSizeBeforeDelete = <%= entityInstance %>Repository.findAll().size();
 

--- a/test-integration/samples/gradle-jwt-react.jdl
+++ b/test-integration/samples/gradle-jwt-react.jdl
@@ -39,6 +39,7 @@ application {
   entities BankAccount, Label, Operation
 }
 
+@service(serviceClass)
 entity BankAccount {
   name String required,
   balance BigDecimal required
@@ -46,6 +47,7 @@ entity BankAccount {
 entity Label {
   label String required minlength(3)
 }
+@service(serviceImpl)
 entity Operation {
   date Instant required,
   description String,

